ui/job-view/pushes/PushList.jsx

```diff
@@ -188,7 +215,7 @@ class PushList extends React.Component {
                 color="darker-secondary"
                 outline
                 className="btn-light-bordered"
-                onClick={() => fetchNextPushes(count)}
+                onClick={() => this.getNextPushes(count)}
```

Comment:
I hit an issue with this because the `fetchNextPushes` prop becauses it calls `addPushes` which uses `replaceLocation`. That way of updating params doesn't hit the `componentDidUpdate` method in job-view App that would trigger an update of FilterModel, which is involved in whether the filters are showing.

Reply:
It looks like it's only updating `fromChange` **before** the fetch of next pushes is complete.  So the range will be off.  It will need to update that param after the fetch is complete.

It also appears that changing the URL by hand causes a page reload, even if you're just modifying the `resultStatus` param.  This was one of the advantages of the hash URLs, since that gave us the ability to decide when to reload the page.  Is there a way to control that with `history'?  I know that you can listen for `history` changes with:

    history.listen((location) => {
        console.log(`You changed the page to: ${location.pathname}`)
    })

But this is the opposite issue of too much page reloading.  Perhaps `react-router-dom` has a way to control that.  Or perhaps we want to actually still use some params after the `#` like flitering to prevent page reload?  

I wonder if you may have to use a listener on `history` like:

    history.listen((location) => {
            console.log(`You changed the page to: ${location.pathname}`)
        })

A big part of using those event listeners was handling when someone changed the URL by hand.  Like changing the `resultStatus` property in this branch will reload the page.

Reply:
yeah, I was really doing it there so we have access to the location props and it was a bit confusing to use two different ways of updating the location. If you do it on the window object directly - which is how it's mostly done in the jobs view probably for the historical reasons you mentioned - it won't trigger a component life cycle update. But I'll take another look at this with fresh eyes :)